srml/staking/src/lib.rs

```diff
@@ -1353,6 +1417,45 @@ impl<T: Trait> OnFreeBalanceZero<T::AccountId> for Module<T> {
 	}
 }
 
+/// Add reward points to block authors:
+/// * 20 points to the block producer for producing a (non-uncle) block in the relay chain,
+/// * 2 points to the block producer for each reference to a previously unreferenced uncle, and
+/// * 1 point to the producer of each referenced uncle block.
+impl<T: Trait + authorship::Trait> authorship::EventHandler<T::AccountId, T::BlockNumber> for Module<T> {
+	fn note_author(author: T::AccountId) {
+		Self::add_reward_points_to_validator(author, 20);
+	}
+	fn note_uncle(author: T::AccountId, _age: T::BlockNumber) {
+		Self::add_reward_points_to_validator(<authorship::Module<T>>::author(), 2);
+		Self::add_reward_points_to_validator(author, 1);
+	}
+}
+
+// This is guarantee not to overflow on whatever values.
+// `num` must be inferior to `den` otherwise it will be reduce to `den`.
+fn multiply_by_rational<N>(value: N, num: u32, den: u32) -> N
```

Comment:
Could be used in the future into support. Looks like something that other modules can also use.

Reply:
do you think having just this functions as is is interesting or should we provide a structure like `RationalU32 { num: u32, den: u32 }`

Reply:
Based on your recommendation I think a `RationalU32 ` with proper `Mul` impls will cover what I recommended. 👍